### PR TITLE
Adds option to replace default Hub logo with custom logo

### PR DESCRIPTION
**Pull request by pratap0007**
User can add their custom logo in place of default hub logo by adding their custom logo's base64 encoded uri data in the UI config

Signed-off-by: Shiv Verma <shverma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [x] Run UI Unit Tests, Lint Checks with `make ui-check`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/image/start.sh [hunk 1]
```diff
@@ -17,6 +17,7 @@ window.config = {
   API_VERSION: '$API_VERSION',
   AUTH_BASE_URL: '$AUTH_BASE_URL',
   REDIRECT_URI: '$REDIRECT_URI',
+  CUSTOMLOGO_BASE64_URI: '$CUSTOMLOGO_BASE64_URI',
```
Comment (PuneetPunamiya) [inline ui/image/start.sh:20]:
```suggestion
  CUSTOM_LOGO_BASE64_URI: '$CUSTOMLOGO_BASE64_URI',
```
🙈

File: ui/src/config/constants.tsx [hunk 2]
```diff
@@ -4,10 +4,12 @@ window.config = window.config || {
   API_URL: 'https://fake.api.hub.tekton.dev',
   API_VERSION: 'v1',
   AUTH_BASE_URL: 'no AUTH_BASE_URL set',
-  REDIRECT_URI: 'no REDIRECT_URI set'
+  REDIRECT_URI: 'no REDIRECT_URI set',
+  CUSTOMLOGO_BASE64_URI: 'no REDIRECT_URI set'
```
Comment (PuneetPunamiya) [inline ui/src/config/constants.tsx:8]: Can you fix the value here please

**Review by PuneetPunamiya — COMMENTED** [review]
Patch looks fine to me, could you please add one more commit to update the config files as well ?

**Review by vdemeester — APPROVED** [review]

File: ui/src/containers/Header/index.tsx [hunk 3]
```diff
@@ -73,7 +73,17 @@ const Header: React.FC = observer(() => {
   return (
     <React.Fragment>
       <PageHeader
-        logo={<Brand src={logo} alt="Tekton Hub Logo" onClick={homePage} />}
+        logo={
+          <Brand
+            src={
+              CUSTOM_LOGO_BASE64_URI != ''
+                ? `data:image/png;base64,${CUSTOM_LOGO_BASE64_URI}`
```
Comment (jkandasa) [inline ui/src/containers/Header/index.tsx:80]: do not force media type to `image/png`, allow custom input.
`CUSTOM_LOGO_BASE64_URI` is not a uri. it is a data.

Comment (PuneetPunamiya) [reply ui/src/containers/Header/index.tsx]: @jkandasa I agree with you, but in that case we might have to restrict with atleast some data types right ?

Comment (PuneetPunamiya) [reply ui/src/containers/Header/index.tsx]: @jkandasa I agree with you, but in that case we might have to restrict with atleast some data types right ?

**Review by vinamra28 — APPROVED** [review]

File: ui/src/containers/Header/index.tsx (hunk 3 above)
Comment (jkandasa) [reply ui/src/containers/Header/index.tsx]: @PuneetPunamiya May be, IMO leaving without a restriction is a fine option. I am not familiar with supported types.

**Review by jkandasa — APPROVED** [review]

File: docs/DEPLOYMENT.md [hunk 4]
```diff
@@ -390,8 +390,14 @@ data:
   API_VERSION: API VERSION  <<< Update this by API version (For e.g-"v1")
   AUTH_BASE_URL: AUTH URL << Update this by auth url
   REDIRECT_URI: UI URL << Update this by ui url
+  CUSTOM_LOGO_MEDIA_TYPE: << Update this by your logo media type
+  CUSTOM_LOGO_BASE64_DATA: << Update this by Base64 encode DATA of Custom Logo
 ```
 
+### Add Custom Logo
+
+Create the Base64 encode value of Custom logo and update the `CUSTOM_LOGO_BASE64_DATA` value in the `config/03-ui/30-ui-configmap.yaml` and also update media type of your logo in `CUSTOM_LOGO_MEDIA_TYEP`, default media type is `image/png` .
```
Comment (jkandasa) [inline docs/DEPLOYMENT.md:399]: typo `CUSTOM_LOGO_MEDIA_TYEP`

**Pull request merged**